src/common/async/yield_waiter.h

```diff
@@ -189,3 +278,23 @@ class yield_waiter<void> {
 };
 
 } // namespace ceph::async
+
+namespace boost::asio {
+
+// forward the handler's associated executor, allocator, cancellation slot, etc
+template <template <typename, typename> class Associator,
```

Comment:
Template-templates on the move! Here, we look to be playing some CRTP games and are very, very happy that auto is a thing. I need to read about Associator, but it is pleasant compared to The Old Way(TM)!

Reply:
> I need to read about Associator

https://www.boost.org/doc/libs/1_87_0/doc/html/boost_asio/overview/model/associators.html might be a good place to start. in asio, it's a common pattern for handlers to wrap other handlers. this template is a convenient way to preserve properties of the wrapped handler